### PR TITLE
CreateAgreement: Write Tests

### PR DESCRIPTION
**Pull request by vahidbazzaz**
PR implements the transfer agreement test cases (4.1) as described in the test plan.

### CODE REVIEW

**Comment by vahidbazzaz** [comment]
[2.0 CreateAgreement: write test](https://trello.com/c/XO2WIRa9)

**Comment by vahidbazzaz** [comment]
@pylipp @HaGuesto I've applied the transfer agreement enum changes to the PR

File: react/src/tests/test-utils.tsx [hunk 1]
```diff
@@ -16,6 +16,11 @@ import {
   ApolloProvider,
   DefaultOptions,
 } from "@apollo/client";
+import {
+  GlobalPreferencesContext,
+  IGlobalPreferencesContext,
+} from "providers/GlobalPreferencesProvider";
+import { organisation1 } from "mocks/oraganisations";
```
Comment (pylipp) [inline react/src/tests/test-utils.tsx:23]: Small typo in the mock filename :)

File: react/src/views/Transfers/CreateTransferAgreement/CreateTransferAgreementView.test.tsx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+  const selectBaseDropDown = screen.getByRole("combobox", { name: /partner bases/i });
+  expect(selectBaseDropDown).toBeInTheDocument();
+  await user.click(selectBaseDropDown);
+  expect(await screen.findByText("Thessaloniki")).toBeInTheDocument();
+  expect(await screen.findByText("Samos")).toBeInTheDocument();
+  const selectedBase = screen.getByRole("button", { name: /samos/i });
+  expect(selectedBase).toBeInTheDocument();
+  await user.click(selectedBase);
+  // Test case 4.1.1.4	- Content: Display Source Organisation name on the label
+  expect(screen.getByText(/boxaid bases/i)).toBeInTheDocument();
```
Comment (pylipp) [inline react/src/views/Transfers/CreateTransferAgreement/CreateTransferAgreementView.test.tsx:114]: This seems to be a duplicate of line 98?

File: react/src/views/Transfers/CreateTransferAgreement/CreateTransferAgreementView.test.tsx [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+  // Test case 4.1.2.5 - The "Valid until" field is optional, but only valid date formats should be entered
+  await selectOptionInSelectField(user, /partner organisation/i, "BoxCare");
+  const transferTypeSendingToLabel = screen.getByLabelText("Sending to");
+  expect(transferTypeSendingToLabel).not.toBeChecked();
+  await user.click(transferTypeSendingToLabel);
+  expect(transferTypeSendingToLabel).toBeChecked();
+
+  const validUntil = screen.getByLabelText(/valid until/i) as HTMLInputElement;
+  const testInavalidValueForValidUntil = addDays(new Date(), -2).toJSON().split("T")[0];
```
Comment (pylipp) [inline react/src/views/Transfers/CreateTransferAgreement/CreateTransferAgreementView.test.tsx:161]: typo: invalid

**Review by pylipp — APPROVED** [review]
Only minor remarks, looks good otherwise =) Thank you!

**Pull request merged**